detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/suppressors/AnnotationSuppressor.kt

```diff
@@ -1,30 +1,65 @@
 package io.gitlab.arturbosch.detekt.core.suppressors
 
 import io.gitlab.arturbosch.detekt.api.ConfigAware
+import io.gitlab.arturbosch.detekt.rules.fqNameOrNull
+import org.jetbrains.kotlin.name.FqName
 import org.jetbrains.kotlin.psi.KtAnnotated
 import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtTypeReference
 import org.jetbrains.kotlin.psi.psiUtil.getStrictParentOfType
+import org.jetbrains.kotlin.resolve.BindingContext
 
-internal fun annotationSuppressorFactory(rule: ConfigAware): Suppressor? {
-    val annotations = rule.valueOrDefault("ignoreAnnotated", emptyList<String>())
+internal fun annotationSuppressorFactory(rule: ConfigAware, bindingContext: BindingContext): Suppressor? {
+    val annotations = rule.valueOrDefault("ignoreAnnotated", emptyList<String>()).map {
+        it.qualifiedNameGlobToRegex()
+    }
     return if (annotations.isNotEmpty()) {
         { finding ->
             val element = finding.entity.ktElement
-            element != null && annotationSuppressor(element, annotations)
+            element != null && annotationSuppressor(element, annotations, bindingContext)
         }
     } else {
         null
     }
 }
 
-private fun annotationSuppressor(element: KtElement, annotations: List<String>): Boolean {
-    return element.isAnnotatedWith(annotations)
+private fun annotationSuppressor(
+    element: KtElement,
+    annotations: List<Regex>,
+    bindingContext: BindingContext
+): Boolean {
+    return element.isAnnotatedWith(annotations, bindingContext)
 }
 
-private fun KtElement.isAnnotatedWith(annotationNames: Iterable<String>): Boolean {
-    return if (this is KtAnnotated && annotationEntries.find { it.typeReference?.text in annotationNames } != null) {
-        true
-    } else {
-        getStrictParentOfType<KtAnnotated>()?.isAnnotatedWith(annotationNames) ?: false
+private fun KtElement.isAnnotatedWith(annotationNames: Iterable<Regex>, bindingContext: BindingContext): Boolean {
+    if (this is KtAnnotated) {
+        val references = annotationEntries.mapNotNull { it.typeReference }
+        if (references.find { it.text in annotationNames } != null) {
```

Comment:
I noticed this pattern multiple times, wouldn't
```suggestion
        if (references.any { it.text in annotationNames }) {
```
be more expressive?

Reply:
That's true! Thanks! I edited it manually because I had that issue twice and I need to fix other minor issues.

This could be a good rule to add in detekt. Use `any` instead of `find`.

Reply:
I created an issue to track this new rule: #4243